src/Raven.Server/Rachis/FollowerAmbassador.cs

```diff
@@ -218,7 +220,19 @@ private unsafe void Run()
                                 }
                                 // either we have new entries to send, or we waited for long enough 
                                 // to send another heartbeat
-                                task.Wait(TimeSpan.FromMilliseconds(_engine.ElectionTimeout.TotalMilliseconds / 3));
+                                var timeToShave = DateTime.UtcNow - timeFromLastSend;
+                                if (_engine.Log.IsInfoEnabled && timeToShave > TimeSpan.FromMilliseconds(50))
+                                {
+                                    _engine.Log.Info($"FollowerAmbassador {_engine.Tag}: waited for {timeToShave.TotalMilliseconds}ms before i started to wait for next heartbeat.");
+                                }
+
+                                task.Wait(TimeSpan.FromMilliseconds(Math.Max(_engine.ElectionTimeout.TotalMilliseconds / 3 - timeToShave.TotalMilliseconds, 0)));
+
+                                var totalTimeWaited = DateTime.UtcNow - timeFromLastSend;
+                                if (_engine.Log.IsInfoEnabled && totalTimeWaited >= TimeSpan.FromMilliseconds(_engine.ElectionTimeout.TotalMilliseconds))
```

Comment:
This will not show up in the errors that the tests raises, so it won't be useful in finding out what the root cause is